homeworks/04/instagram-api/src/main/java/com/example/instagramapi/PostsServlet.java

```diff
@@ -0,0 +1,290 @@
+package com.example.instagramapi;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+
+
+@WebServlet(name = "PostsServlet", urlPatterns = "/posts/*", loadOnStartup = 1)
+public class PostsServlet extends HttpServlet {
+    @Override
+    public void init() throws ServletException {
+        getServletConfig().getServletContext().setAttribute("database", new HashMap<String, User>());
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String pathInfo = req.getPathInfo();
+        HashMap<String, User> users = (HashMap<String, User>) getServletConfig().getServletContext().getAttribute("database");
+        String currentUser = (String) getServletConfig().getServletContext().getAttribute("profile");
+
+        if(currentUser == null){
+            resp.setStatus(401);
+            resp.getWriter().println("Please log in before posting/commenting/liking!");
+            return;
+        }
+
+        if(pathInfo == null){
```

Comment:
The path for posting an image is "/posts/". You should make another servlet for handling requests for posts. The "/path/*" URL should be used for all likes and comments request.